src/PIL/Image.py

```diff
@@ -2211,13 +2228,9 @@ def resize(self, size, resample=None, box=None, reducing_gap=None) -> Image:
             msg = "reducing_gap must be 1.0 or greater"
             raise ValueError(msg)
 
-        size = tuple(size)
-
         self.load()
         if box is None:
             box = (0, 0) + self.size
-        else:
-            box = tuple(box)
```

Comment:
@srittau @radarhere This change ended up being a breaking change for me when upgrading from 1.3.0 to 1.4.0 since `.resize()` previously would accept a numpy array as size. This now fails with an internal error message:
```python
  File "/usr/local/lib/python3.11/site-packages/PIL/Image.py", line 2297, in resize
    if self.size == size and box == (0, 0) + self.size:
       ^^^^^^^^^^^^^^^^^
ValueError: The truth value of an array with more than one element is ambiguous. Use a.any() or a.all()
```
If this change was intentional I think it would be best to raise a nice error message or revert this and relax the input type requirements.

Reply:
Let's move this conversation to #8195

Reply:
https://github.com/python-pillow/Pillow/pull/8201 has now restored the ability to use a NumPy array.